### PR TITLE
Show all columns in dataset management by default

### PR DESCRIPTION
**Pull request by allisonking**
Closes https://github.com/ethyca/fides/issues/846

### Code Changes

* [x] Remove filter which made all but one column show
* [x] Adds tests for column dropdown behavior

Also beyond the scope of this ticket, but also:
* [x] Update menu behavior to match that of menus in the datamap UI (Elliot and I interpreted the design differently, and after using the datamap UI, I felt like he was right! see videos at bottom for the difference)

### Steps to Confirm

* [ ] Play around with the Column dropdown and the data category dropdown

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* Documentation Updated:
  - [ ] documentation complete, or draft/outline provided (tag docs-team to complete/review on this branch)
  - [ ] documentation issue created (tag docs-team to complete issue separately)
* [x] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [x] Update `CHANGELOG.md`

### Description Of Changes

All columns are rendered by default (tiniest change https://github.com/ethyca/fides/commit/859a983d6738bd05292856b5333fb692b1bc6910)

image

The previous menu behavior kept an internal state and would only apply it when clicking 'Done'

https://user-images.githubusercontent.com/24641006/178800681-684bcd3b-ad53-47c0-a168-9ee684e53b33.mov

I think it makes more sense to match the datamap UI behavior, where checking boxes changes state immediately, and the "Done" button closes the menu

https://user-images.githubusercontent.com/24641006/178800830-28262110-caee-41af-aabd-07e0f1edf246.mov

### CODE REVIEW

**Review by ThomasLaPiana — APPROVED** [review]
LGTM!

**Comment by ThomasLaPiana** [comment]
@allisonking I agree that the instant state change makes more sense, good change!

**Review by ssangervasi — APPROVED** [review]

File: clients/admin-ui/src/features/dataset/ColumnDropdown.tsx [hunk 1]
```diff
@@ -20,19 +19,23 @@ interface Props {
...
 
-  const handleSubmit = () => {
+  const handleChange = (column: ColumnMetadata) => {
+    let newColumns: ColumnMetadata[] = [];
+    if (selectedColumns.filter((col) => col.name === column.name).length > 0) {
+      // remove from selected columns
+      newColumns = selectedColumns.filter((col) => col.name !== column.name);
+    } else {
+      newColumns = [...selectedColumns, column];
+    }
```
[the remark this answers is not in the document]
Comment (allisonking) [reply clients/admin-ui/src/features/dataset/ColumnDropdown.tsx:33]: oh that's neat, I'll try it out! thanks for the tip 🧠

[the remark this answers is not in the document]
Comment (allisonking) [reply clients/admin-ui/src/features/dataset/ColumnDropdown.tsx]: lemme know if this looks right? https://github.com/ethyca/fides/pull/898/commits/91f43976aeca8494eeea3e3868f94677528a95d8

**Pull request merged**